src/library/scala/collection/View.scala

```diff
@@ -364,7 +364,12 @@ object View extends IterableFactory[View] {
         i += 1
```

Comment:
I'm wondering if `next()` ought to throw an `IndexOutOfBoundsException` rather than a `NoSuchElementException`

Reply:
I have to disagree, for two reasons: 
- The `next()` method is declared in the  `Iterable` super-trait, and it has to be compatible with the super-type interface.
- The `View` trait covers collections without any notion of indexes, like key-value maps.